.github/workflows/cran-emulator.yml

```diff
@@ -0,0 +1,98 @@
+name: CRAN Emulator
+
+
+on:
+  push:
+    tags:
+      - "v*"
+  workflow_call:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+
+
+concurrency:
+  group: cran-systems-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: ${{ matrix.system }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        system:
+          - ghcr.io/insightsengineering/debian-gcc-devel:latest
+          - ghcr.io/insightsengineering/debian-gcc-devel:latest
+          - ghcr.io/insightsengineering/fedora-clang-devel:latest
+          - ghcr.io/insightsengineering/fedora-gcc-devel:latest
+          - ghcr.io/insightsengineering/debian-gcc-patched:latest
+          - ghcr.io/insightsengineering/debian-gcc-release:latest
+
+    container:
+      image: ${{ matrix.system }}
+      # All options: https://docs.docker.com/engine/reference/commandline/create/#options
+      options: --user root
+
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      # Specs from https://www.stats.ox.ac.uk/pub/bdr/Rconfig/r-devel-linux-x86_64-fedora-clang
+      _R_CHECK_INSTALL_DEPENDS_: true
+      _R_CHECK_SUGGESTS_ONLY_: true
+      _R_CHECK_NO_RECOMMENDED_: true
+      _R_CHECK_DOC_SIZES2_: true
+      _R_CHECK_DEPRECATED_DEFUNCT_: true
+      _R_CHECK_SCREEN_DEVICE_: warn
+      _R_CHECK_REPLACING_IMPORTS_: true
+      _R_CHECK_TOPLEVEL_FILES_: true
+      _R_CHECK_DOT_FIRSTLIB_: true
+      _R_CHECK_RD_LINE_WIDTHS_: true
+      _R_CHECK_S3_METHODS_NOT_REGISTERED_: true
+      _R_CHECK_OVERWRITE_REGISTERED_S3_METHODS_: true
+      _R_CHECK_CODE_USAGE_WITH_ONLY_BASE_ATTACHED_: true
+      _R_CHECK_NATIVE_ROUTINE_REGISTRATION_: true
+      _R_CHECK_FF_CALLS_: registration
+      _R_CHECK_PRAGMAS_: true
+      _R_CHECK_COMPILATION_FLAGS_: true
+      _R_CHECK_R_DEPENDS_: true
+      _R_CHECK_PACKAGES_USED_IN_TESTS_USE_SUBDIRS_: true
+      _R_CHECK_SHLIB_OPENMP_FLAGS_: true
+      _R_CHECK_CODE_ASSIGN_TO_GLOBALENV_: true
+      _R_CHECK_CODE_DATA_INTO_GLOBALENV_: true
+      _R_CHECK_PKG_SIZES_: true
+      _R_CHECK_LIMIT_CORES_: true
+      _R_S3_METHOD_LOOKUP_BASEENV_AFTER_GLOBALENV_: true
+      _R_CHECK_AUTOCONF_: true
+      _R_CHECK_THINGS_IN_CHECK_DIR_: true
+      _R_CHECK_THINGS_IN_TEMP_DIR_: true
+      _R_CHECK_THINGS_IN_TEMP_DIR_EXCLUDE_: "^ompi"
+      _R_CHECK_BASHISMS_: true
+      _R_CHECK_ORPHANED_: true
+      _R_CHECK_DEPENDS_ONLY_DATA_: true
+      _R_CHECK_XREFS_PKGS_ARE_DECLARED_: true
+      _R_CHECK_MATRIX_DATA_: true
+      _R_CHECK_RD_VALIDATE_RD2HTML_: true
+      _R_CHECK_RD_MATH_RENDERING_: true
+      _R_CHECK_FORCE_SUGGESTS_: false
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Install devtools
+        run: install.packages("devtools", repos = "https://cloud.r-project.org")
+        shell: Rscript {0}
+
+      - name: Install package dependencies
+        run: devtools::install_dev_deps()
```

Comment:
Please note that this might have undesired side effect on the R CMD CHECK you are executing here. What if your utils package (regardless if it's `devtools` or `pak`) will install also package X and this package will be required but not specified in your tested package. You will have a false positive.
There are two ways to address that problem
A) utils package to temp libpath
- create temp libpath and install utils package there
- use it from temp libpath and install dev deps to default libpath
- RCMDCHECK from default libpath
B) dev deps to temp libpath
- install utils package inside default libpath
- use it from default libpath and install dev deps to temp libpath
- RCMDCHECK from temp libpath
(I personally used B) in verdepcheck and when executing R CMD CHECK via `rcmdcheck` package and stopped on https://github.com/r-lib/rcmdcheck/issues/195. Haven't tried this from shell)